src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -705,6 +705,17 @@ class Peer(
         return incomingPaymentHandler.createInvoice(paymentPreimage, amount, description, extraHops, expirySeconds)
     }
 
+    /** Creates a custom offer and register it with the `offerManager`.
+     *  @param secret A random private key for creating the blinded path of the offer. Must be unique to this offer.
+     *  The offer returned is deterministic, if you need to persist you just need to save the parameters used to create it.
+     */
+    fun createOffer(secret: PrivateKey, amount: MilliSatoshi?, description: String?): OfferTypes.Offer {
```

Comment:
```suggestion
    fun createOffer(secret: PrivateKey, amount: MilliSatoshi?, description: String?): OfferTypes.Offer {
```

Do you think that we should add a sanity check in here when the amount and the description are `null`?

I am assuming that we want to keep using the "default" offer when the amount and the description are not specified - or there is any other use case where we want to have different default offers?